mlflow/utils/proto_json_utils.py

```diff
@@ -239,6 +239,11 @@ def cast_df_types_according_to_schema(pdf, schema):
                     # The conversion will be done in `_enforce_schema` while
                     # `PyFuncModel.predict` being called.
                     pass
+                elif isinstance(col_type_spec, (Array, Object)):
+                    # for array and object colspec types, we expect
+                    # the input to be list or dictionary, so we shouln't
+                    # convert it here
+                    pass
```

Comment:
Question: for Array/Object schema, should we support passing pandas dataframe as input?

Reply:
We should support what is already supported. For new schema's in input_example, we could validate it directly without converting to pandas dataFrame.